src/rules/html-self-closing.ts

```diff
@@ -27,32 +27,65 @@ export default createRule("html-self-closing", {
     },
     schema: [
       {
-        type: "object",
-        properties: {
-          void: {
-            enum: ["never", "always", "ignore"],
+        oneOf: [
+          {
+            properties: {
+              void: {
+                enum: ["never", "always", "ignore"],
+              },
+              normal: {
+                enum: ["never", "always", "ignore"],
+              },
+              component: {
+                enum: ["never", "always", "ignore"],
+              },
+              svelte: {
+                enum: ["never", "always", "ignore"],
+              },
+            },
+            additionalProperties: false,
           },
-          normal: {
-            enum: ["never", "always", "ignore"],
+          {
+            enum: ["all", "html", "none"],
           },
-          component: {
-            enum: ["never", "always", "ignore"],
-          },
-          svelte: {
-            enum: ["never", "always", "ignore"],
-          },
-        },
-        additionalProperties: false,
+        ],
       },
     ],
   },
   create(ctx) {
-    const options = {
+    let options = {
       void: "always",
       normal: "always",
       component: "always",
       svelte: "always",
-      ...ctx.options?.[0],
+    }
+
+    if (typeof ctx.options?.[0] === "object") {
```

Comment:
What is `ctx.options?.[0] === null`?

Reply:
that is "object", however
```js
{
  test: 1,
  ...null,
}
// is
{
  test: 1,
}
```

Reply:
Then the reassignment is unnecessary, right? And it can be handled in the following `default` `case`, no need to handle it here?